alphabase/smiles/smiles.py

```diff
@@ -0,0 +1,116 @@
+from typing import Optional
+
+import pandas as pd
+from rdkit import Chem
+from rdkit.Chem.rdmolops import ReplaceSubstructs, SanitizeMol
+
+from alphabase.constants.aa import AA_Formula
+from alphabase.constants.modification import MOD_DF
+
+N_TERM_PLACEHOLDER = "[Xe]"
+C_TERM_PLACEHOLDER = "[Rn]"
+
+MOD_DF_SMILES = MOD_DF[MOD_DF["smiles"] != ""]
+
+aa_smiles = {
```

Comment:
I know that import-level logic is sort of a thing around here :-D
Could we turn all this into a class, and do these operations only when needed?

Reply:
I agree. I think it was already suboptimal for the registry pattern and more so for more involved calculations like `Chem.MolFromSmiles()`

Reply:
Rewrote this into a class